libass/ass_render.c

```diff
@@ -1509,11 +1506,13 @@ static void measure_text(ASS_Renderer *render_priv)
         max_desc = FFMAX(max_desc, cur->desc);
         if (cur->symbol != '\n' && cur->symbol != 0)
             scale = 1.0 / 64;
+        if (i == text_info->length - 1 || text_info->glyphs[i + 1].linebreak) {
```

Comment:
That's avoiding a code duplication at the cost of a logic complication. I think it's better to move common code into a separate function instead.

Reply:
Done.